server/jetstream_cluster.go

```diff
@@ -2008,6 +2011,8 @@ func (js *jetStream) monitorStream(mset *stream, sa *streamAssignment, sendSnaps
 			// If we were successful lookup up our stream now.
 			if err == nil {
 				if mset, err = acc.lookupStream(sa.Config.Name); mset != nil {
+					mset.monitorWg.Add(1)
+					defer mset.monitorWg.Done()
```

Comment:
So even if - not sure if that is possible - we enter with mset != nil and end-up in this code, the number of Add() and Done() would match, so that is ok!

Reply:
basically